src/main/java/uk/gov/hmcts/reform/opal/model/FileSequence.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.reform.opal.model;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.opal.model.dto.StandardBankingFileName;
+
+import java.util.HashMap;
+
+@Component
+public class FileSequence {
+
+    private HashMap<String, String> sequenceMap = new HashMap<>();
+
+    public String getAndIncrementSequence(StandardBankingFileName fileName) {
```

Comment:
Could this be accessed by multiple threads?
Should we have some concurrency protection?
I may have missed something here.

Reply:
Im not sure why I marked the class as a component. Each job that needs to sequence its input files will create a new instance of this FileSequence class.